### ISSUE TITLE
test_data

### ISSUE THREAD

**Issue by mauimoana** [zekarias-tilahun/cas2vec#2]
Hello, I noticed that you did not include the test evaluation in your code. Would this have a significant impact on the results?

Thanks for your time.
Best wishes

**Comment by zekarias-tilahun** [comment]
Hello, I'm not sure if I understood your question correctly, and it's been a while since I wrote the code.

If your question is where is the test data, after a quick check at the code I see that the test data is automatically created from the training data. That is, for a given cascade in the ``train.txt`` file, up to a certain time/length it will be used as a training data the rest will be used for evaluation (test).

**Comment by mauimoana** [comment]
Thanks very much for your reply.  Your code did include evaluation, but it does not use the test set. I added the test part in your codes, but the results seem to be bad. In my opinion, the reason for this might be the undersampling strategy, which might distort the distribution of the data. The distribution of the test set is different from that of the training and validation sets. 
Thanks very much for your time.
Best wishes.

**Comment by zekarias-tilahun** [comment]
Yes, the method is not tested under a different distribution than the training data.
